joomla/joomla-cms
# [4.0] Decouple messages from core.js

Pull Request for Issue # .

### Summary of Changes
- This PR moves the core.js logic for rendering messages to another file
- It uses import for bringing in the alert custom element (one less HTTP request)

### Why?

Well, there is no way to override this part of the code without overriding the whole core.js, which is terrible as this is supposed to be the javascript API of the CMS. Bad architecture.

### Depends/extends https://github.com/joomla/joomla-cms/pull/32740

### Testing Instructions
- Apply the PR
- try to login with the wrong credential in the front end and back end (should get an alert)
- Try to upload an image in the media manager (you should also get an alert)


### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request



### Documentation Changes Required

Not really, the code is the same, we only moved it to an own file that is included from the respective layout (thus is properly overridable now)
